### PR TITLE
{vm-repair} Migrate commands calling Compute module to aaz-based implementation and update deprecated package

### PR DESCRIPTION
**Pull request by william051200**
### 🤖 PR Validation — ️✔️ All clear

| Breaking Changes |
| :--: |
| ️✔️ None |

Migrate code from Azure SDK to AAZ based commands for compute operations
Replace deprecated pkgutil loader with importlib.import_module

After running some tests for version compatibility, here are the results I got,
az extension add vm-repair 2.2.1 package generated from azdev 0.2.12 & 0.2.13 will show version = null

Those successfully installed (with no null version) also runs properly, with no error.
This confirms the migration has no issue & the issue raised by user is caused by version incompatibility.

```
Azure CLI 2.87:
1. azdev 0.2.13 - fail (no version)
2. azdev 0.2.12 - fail (no version)
3. azdev 0.2.10 - success
4. azdev 0.2.9 - success
5. azdev 0.2.8 - success

Azure CLI 2.88:
1. azdev 0.2.13 - success
2. azdev 0.2.12 - success
3. azdev 0.2.10 - success
4. azdev 0.2.9 - success
5. azdev 0.2.8 - success
```
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install azdev` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
vm-repair

**Review by JaysonTaiMicrosoft — DISMISSED** [review]

File: src/vm-repair/azext_vm_repair/repair_utils.py [hunk 1]
```diff
@@ -650,8 +647,8 @@ def _fetch_compatible_windows_os_urn(source_vm, source_vm_instance_view):
 
     logger.debug('Fetched Urns:\n%s', urns)
     # temp fix to mitigate Windows disk signature collision error
-    os_image_ref = source_vm.storage_profile.image_reference
-    if os_image_ref and isinstance(os_image_ref.version, str) and os_image_ref.version in urns[0]:
+    os_image_ref = source_vm.get('storageProfile', {}).get('imageReference', {})
+    if os_image_ref and os_image_ref.get('version') in urns[0]:
```
Comment (EdwinBernal1) [inline src/vm-repair/azext_vm_repair/repair_utils.py]: The AAZ migration removes the previous string guard around the image version. For specialized or custom image references, `imageReference` may be nonempty while `version` is absent or null, so `None in urns[0]` raises `TypeError` and aborts Windows repair creation. Please restore the guard, for example:

```suggestion
    version = os_image_ref.get('version')
    if isinstance(version, str) and version in urns[0]:
```

Please also add a regression test for `version=None`.

**Review by EdwinBernal1 — CHANGES_REQUESTED** [review]
The AAZ migration and Azure CLI 2.83.0 compatibility gate look consistent, and CI is green. I found one blocking regression in Windows repair image selection: a missing or null `imageReference.version` now raises `TypeError`. Please restore the prior string guard and add a focused regression test before merging.

**Review by EdwinBernal1 — APPROVED** [review]
